source/dubregistry/internal/utils.d

```diff
@@ -1,15 +1,86 @@
 module dubregistry.internal.utils;
 
+import vibe.core.core;
+import vibe.core.concurrency;
+import vibe.core.task;
 import vibe.inet.url;
+import vibe.inet.path;
+
+import std.algorithm : any;
+import std.file : tempDir;
+import std.path;
+import std.process;
 
 URL black(URL url)
 {
-	if (url.username.length > 0) url.username = "***";
-	if (url.password.length > 0) url.password = "***";
+	if (url.username.length > 0)
+		url.username = "***";
+	if (url.password.length > 0)
+		url.password = "***";
 	return url;
 }
 
 string black(string url)
 {
 	return black(URL(url)).toString();
 }
+
+static immutable string logoOutputFolder = "uploads/logos";
+
+static immutable string[] logoFormats = [".webp", ".png"];
+
+/// 
+/// Params:
+///   file = the file to convert
+///   name = the logo name to put in uploads/logos
+///   deleteExisting = if false, throw an exception if the files already exist
+///   deleteFinish = if true, delete the input file after at least one successful conversion
+auto generateLogo(Path file, string name, bool deleteExisting = false, bool deleteFinish = true)
+{
+	if (!name.length)
+		throw new Exception("name may not be empty");
+	foreach (format; logoFormats)
+		if (existsFile(buildPath(logoOutputFolder, name ~ format)))
+		{
+			if (deleteExisting)
+				removeFile(buildPath(logoOutputFolder, name ~ format));
+			else
+				throw new Exception("logo " ~ format ~ " already exists");
+		}
+	auto t = runWorkerTaskH(&generateLogoUnsafe, Task.getThis(), file, name);
+	auto success = receiveOnlyCompat!(bool[logoFormats.length]);
+	t.join();
```

Comment:
Do you need to wait for this? Isn't the idea of using a separate worker to avoid blocking the response sent to the client?

Reply:
well it closes after sendCompat anyway so I wanted to make sure that the task exited. Not sure about the internals, thought that maybe the same as with zombie processes could happen if I didn't.